pos_order_reorder/static/src/js/Screens/TicketScreen/ControlButtons/ReorderButton.esm.js

```diff
@@ -0,0 +1,67 @@
+/** @odoo-module **/
+
+import PosComponent from "point_of_sale.PosComponent";
+import Registries from "point_of_sale.Registries";
+import {Orderline} from "point_of_sale.models";
+
+class ReorderButton extends PosComponent {
+    get isEmptyOrder() {
+        if (!this.props.order) return true;
+        return this.props.order.is_empty();
+    }
+    _reOrder() {
+        if (this.isEmptyOrder) {
+            return;
+        }
+        const order = this.props.order;
+        const pos = this.env.pos;
+        const partner = order.get_partner();
+        const newOrder = pos.add_new_order();
+        if (partner) {
+            newOrder.set_partner(partner);
+        }
+        if (order.fiscal_position) {
+            newOrder.fiscal_position = order.fiscal_position;
+        }
+        if (order.pricelist) {
+            newOrder.set_pricelist(order.pricelist);
+        }
+        const lines = order.get_orderlines();
+        for (var i = 0; i < lines.length; i++) {
+            const line = lines[i];
```

Comment:
```suggestion
        for (const line of order.get_orderlines()) {
```

Reply:
Could you explain in more detail why this change is necessary?

Reply:
It's not necessary, just a suggestion to write modern js ;)